clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -150,6 +152,7 @@ private void maybePropagateMetadataError() {
         try {
             metadata.maybeThrowAnyException();
         } catch (Exception e) {
+            metadataError = Optional.of(e);
             backgroundEventHandler.add(new ErrorEvent(e));
```

Comment:
Seems to me you're really close to being able to remove this duplicated way of propagating metadata errors (and simplify the `isPassedByErrorEvent`)

We now have a way to propagate metadata errors based on the triggering event, to fill a gap we have to let calls like position know about them. Then why exactly is it that we still need to propagate them via ErrorEvents still too? Seems to be for the consumer.poll case only, but I would expect that (once the gap from the comment above is addressed), the CheckAndUpdatePositions event should fail with subscription metadata errors, no matter if called from consumer.position or from consumer.poll. Makes sense?